server/src/test/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilderTests.java

```diff
@@ -837,21 +837,22 @@ public void testCacheability() throws IOException {
         assertEquals("query should " + (isCacheable ? "" : "not") + " be cacheable: " + queryBuilder.toString(), isCacheable,
                 context.isCacheable());
 
-        // check the two non-cacheable cases explicitly
         ScoreFunctionBuilder<?> scriptScoreFunction = new ScriptScoreFunctionBuilder(
                 new Script(ScriptType.INLINE, MockScriptEngine.NAME, "1", Collections.emptyMap()));
-        RandomScoreFunctionBuilder randomScoreFunctionBuilder = new RandomScoreFunctionBuilderWithFixedSeed();
-
-        for (ScoreFunctionBuilder<?> scoreFunction : List.of(scriptScoreFunction, randomScoreFunctionBuilder)) {
-            FilterFunctionBuilder[] functions = new FilterFunctionBuilder[] {
-                    new FilterFunctionBuilder(RandomQueryBuilder.createQuery(random()), scoreFunction) };
-            queryBuilder = new FunctionScoreQueryBuilder(functions);
+        queryBuilder = new FunctionScoreQueryBuilder(new FilterFunctionBuilder[] {
+            new FilterFunctionBuilder(RandomQueryBuilder.createQuery(random()), scriptScoreFunction) });
+        context = createShardContext();
+        rewriteQuery = rewriteQuery(queryBuilder, new QueryShardContext(context));
+        assertNotNull(rewriteQuery.toQuery(context));
+        assertTrue("script should query should be cacheable: " + queryBuilder.toString(), context.isCacheable());
```

Comment:
"script should query should be cacheable: "  -> "function script query should be cacheable"

Reply:
Added.